pydantic/main.py

```diff
@@ -577,11 +589,15 @@ def _iter(
     ) -> TupleGenerator:
         # Merge field set excludes with explicit exclude parameter with explicit overriding field set options.
         # The extra "is not None" guards are not logically necessary but optimizes performance for the simple case.
-        # if exclude is not None or self.__exclude_fields__ is not None:
-        #     exclude = _utils.ValueItems.merge(self.__exclude_fields__, exclude)
-        #
-        # if include is not None or self.__include_fields__ is not None:
-        #     include = _utils.ValueItems.merge(self.__include_fields__, include, intersect=True)
+        if exclude is not None:
```

Comment:
Uncommenting these lines (with some modifications) allowed me to get tests passing, and this method seems to _only_ be used in (deprecated) `copy`.

I tried changing `copy` method to just do:
```
        data = self.model_dump(include=include, exclude=exclude, round_trip=True)
        data = {**data, **(update or {})}
        return self.model_validate(data)
```
but this doesn't work because 1, it actually validates, which makes many things not work with include/exclude, and 2, it doesn't actually do shallow copies.

For now, I'd be inclined to keep the logic like this (since all unit tests pass) and plan to remove `_iter` at the same time we remove `copy`.